website/src/pages/tutorial.md

```diff
@@ -0,0 +1,530 @@
++            })
++        };
+
+        html! {
+-            <p>{format!("{}: {}", video.speaker, video.title)}</p>
++            <p onclick={on_video_select}>{format!("{}: {}", video.speaker, video.title)}</p>
+        }
+    }).collect()
+}
+```
+
+Next, we need to modify the usage of `VideosList` to pass that callback. But before doing that, we should create
+a new component, `VideoDetails`, component that is displayed when a video is clicked.
+
+```rust
+#[derive(Clone, Properties, PartialEq)]
+struct VideosDetailsProps {
+    video: Video,
+}
+
+#[function_component(VideoDetails)]
+fn video_details(VideosDetailsProps { video }: &VideosDetailsProps) -> Html {
+    html! {
+        <div>
+            <h3>{ video.title.clone() }</h3>
+            <img src="https://via.placeholder.com/640x360.png?text=Video+Player+Placeholder" alt="video thumbnail" />
+        </div>
+    }
+}
+```
+
+Now, modify the `App` component to display `VideoDetails` component whenever a video is selected.
+
+```rust {4,6-11,13-15,24-29}
+#[function_component(App)]
+fn app() -> Html {
+    // ...
++    let selected_video = use_state(|| None);
+
++    let on_video_select = {
++        let selected_video = selected_video.clone();
++        Callback::from(move |video: Video| {
++            selected_video.set(Some(video))
++        })
++    };
+
++    let details = selected_video.as_ref().map(|video| html! {
++        <VideoDetails video={video.clone()} />
++    });
+
+    html! {
+        <>
+            <h1>{ "RustConf Explorer" }</h1>
+            <div>
+                <h3>{"Videos to watch"}</h3>
+                <VideosList videos={videos} on_click={on_video_select.clone()} />
+            </div>
++            { for details }
+-            <div>
+-                <h3>{ "John Doe: Building and breaking things" }</h3>
+-                <img src="https://via.placeholder.com/640x360.png?text=Video+Player+Placeholder" alt="video thumbnail" />
+-            </div>
+-        </>
+    }
+}
+```
+
+Don't worry about the `use_state` right now, we'll come back to it later. 
+Note the trick we pulled with `{ for details }`. `Option<_>` implements `Iterator` so we can use it to display the only
+element returned by the `Iterator` with the `{ for ... }` syntax.
+
+### Handling state
+
+Remember the `use_state` used earlier? That is a special function, called a "hook". Hooks are used to "hook" into 
+lifecycle of a function component and perform actions. You can learn more about this hook, and others 
+[here](/next/concepts/function-components/pre-defined-hooks#use_state)
+
+:::note
+Struct components act differently. See the documentation to learn about those.
+:::
+
+## Fetching data (using external REST API)
+
+In a real world application, data will basically never be hardcoded but instead will come from an API. Let's fetch our 
+videos list from external source. For this we will need to add the following crates:
+- [`reqwasm`](https://crates.io/crates/reqwasm)
```

Comment:
(Side note: nothing against `reqwasm` but it would certainly be nice if we could suggest using a more "official" library. Since you are also a maintainer for `gloo`, maybe something like `reqwasm` could be ported over?)

Reply:
I would love to but I'm holding off because I can't publish gloo. I don't have permissions on crates.io